components/NavBar.vue

```diff
@@ -107,37 +107,27 @@
 </template>
 
 <script lang="ts">
-import { useApolloClient, useMutation } from '@vue/apollo-composable'
+import { useApolloClient } from '@vue/apollo-composable'
 import { defineComponent, ref, watch } from '@vue/composition-api'
 import { useRouter } from '#app'
-import { gql } from '~/apollo'
 import { useUiStore } from '~/store'
 import { cacheCurrentUser } from '~/apollo/cache'
+import { useLogoutMutation } from '~/types/graphql'
 
 export default defineComponent({
   setup() {
     const { resolveClient } = useApolloClient()
 
-    const { mutate: logout, onDone } = useMutation(
-      gql(/* GraphQL */ `
-        mutation Logout {
-          logout {
-            result
-          }
-        }
-      `),
-      {
-        update(_cache, _data) {
-          cacheCurrentUser(null)
-        },
-      }
-    )
-    const router = useRouter()
-    onDone(() => {
-      // Reset graphql cache
-      void resolveClient().clearStore()
+    const { mutate: logout, loading, error, onDone } = useLogoutMutation()
 
-      void router.push({ name: 'index' })
+    const router = useRouter()
+    onDone((result) => {
+      if (result.data?.logout) {
+        // Reset graphql cache
+        void resolveClient().clearStore()
+        cacheCurrentUser(null)
```

Comment:
Is this really the "correct" way to handle the cache? Normally, apollo has these update methods that work slightly different then the onDone callers.

Reply:
Actually, Updating the cache here is kind of unnecessary. This would be helpful if the data fetched by me query is the same as the data we get from login mutation(i.e., if we only query for email or id in user field with me query, then we are avoiding a request to the server). But right after the login, we query for user document with me query, which will result in a cache miss, and apollo client will send another request to the server. Let me know if I'm missing something, but I don't think we are avoiding any requests here.

Reply:
I agree with your analysis for what happens at the login. However, here we logout the user and need to tell graphql that this means `user = null` (otherwise the apollo client accesses the cache and thinks the user is still logged in).

Reply:
And even for the login, there are a few requests that profit from caching such as this one here:
https://github.com/JabRef/JabRefOnline/blob/b279d4a1e18e93d5d0670fb550224e7b00948f2c/plugins/authenticated.ts#L19-L27
(which is run on every page load to check if the user is logged in).